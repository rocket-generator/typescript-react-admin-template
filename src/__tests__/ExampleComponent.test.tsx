import { render } from "@testing-library/react"

type Props = {
  message: string
}

const ExampleComponent = ({ message }: Props): JSX.Element => {
  return (
    <div>
      <p>Welcome to React!</p>
      <p>This is an example of a React snapshot test.</p>
      <p>{message}</p>
    </div>
  )
}

describe("example", () => {
  it("renders correctly", () => {
    const rendered = render(<ExampleComponent message="wow" />)
    expect(rendered.asFragment()).toMatchSnapshot()
  })
})
