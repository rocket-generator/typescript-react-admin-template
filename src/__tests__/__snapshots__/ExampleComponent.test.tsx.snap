// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`example > renders correctly 1`] = `
<DocumentFragment>
  <div>
    <p>
      Welcome to React!
    </p>
    <p>
      This is an example of a React snapshot test.
    </p>
    <p>
      wow
    </p>
  </div>
</DocumentFragment>
`;
